src/basic/stat-util.c

```diff
@@ -67,20 +67,33 @@ int is_device_node(const char *path) {
         return !!(S_ISBLK(info.st_mode) || S_ISCHR(info.st_mode));
 }
 
-int dir_is_empty(const char *path) {
-        _cleanup_closedir_ DIR *d;
+int dir_is_empty_at(int dirfd, const char *path) {
```

Comment:
please don't call the parameter dirfd(), after all that's a symbol exposed by glibc, and we should try to avoid such name clashes even if in this case you get away with it. maybe dir_fd or so...